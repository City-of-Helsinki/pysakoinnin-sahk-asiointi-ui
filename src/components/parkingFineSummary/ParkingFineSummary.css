.summary-container {
  display: grid;
  grid-template-columns: 510px auto auto;
  grid-auto-rows: auto;
  grid-gap: 12px 12px;
  margin-top: 50px;
}

.info-field {
  max-width: 384px;
  grid-row-end: auto;
}

.sum-field {
  grid-column-end: 2;
}

.wide-field {
  grid-column-start: 1;
  grid-column-end: span 2;
}

textarea {
  resize: none !important;
}

.vehicle-details-container {
  grid-column-start: 3;
  grid-row: span 13;
}

hr {

  height: 1px;
  width: 100%;
  grid-column-start: 1;
  grid-column-end: span 2;
}
