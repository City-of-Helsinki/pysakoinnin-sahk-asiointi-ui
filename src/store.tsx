import { configureStore } from '@reduxjs/toolkit';
import formContentReducer from './components/formContent/formContentSlice';
import formStepperReducer from './components/formStepper/formStepperSlice';
import extendDueDateFormReducer from './components/extendDueDate/extendDueDateFormSlice';
import userReducer from './components/user/userSlice';

const store = configureStore({
  reducer: {
    formContent: formContentReducer,
    formStepper: formStepperReducer,
    extendDueDateForm: extendDueDateFormReducer,
    user: userReducer
  }
});

export default store;
export type RootState = ReturnType<typeof store.getState>;
