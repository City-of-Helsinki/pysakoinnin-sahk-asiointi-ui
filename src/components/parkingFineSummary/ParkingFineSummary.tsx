import React, { FC, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { setSubmitDisabled } from '../formContent/formContentSlice';
import { formatDate, formatDateTime } from '../../utils/helpers';
import { FoulData, Foul } from '../../interfaces/foulInterfaces';
import Barcode from '../barcode/Barcode';
import './ParkingFineSummary.css';

interface FoulRowProps {
  foul: Foul;
}

const FoulRow: FC<FoulRowProps> = ({ foul }) => {
  const { t } = useTranslation();
  return (
    <>
      <div className="info-field">
        <label>{t('common:fine-info:fine-title')}</label>
        <p>{foul.description}</p>
      </div>
      <div className="info-field">
        <label>{t('common:fine-info:fine-details')}</label>
        <p>{foul.additionalInfo}</p>
      </div>
    </>
  );
};

interface Props {
  foulData: FoulData;
}

const ParkingFineSummary: FC<Props> = ({ foulData }) => {
  const { t } = useTranslation();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(setSubmitDisabled(false));
  }, [dispatch]);

  return (
    <div className="summary-body">
      <div data-testid="parkingFineSummary" className="summary-container">
        <h2 className="show-on-print" aria-hidden="true">
          {t('parking-fine:fine-info')}
        </h2>
        <div className="info-field">
          <label>{t('common:fine-info:fine-timestamp')}</label>
          <p>{formatDateTime(foulData.foulDate)}</p>
        </div>
        <div className="info-field">
          <label>{t('common:fine-info:ref-number:label')}</label>
          <p>{foulData.foulNumber}</p>
        </div>

        <hr />

        <div className="info-field">
          <label>{t('common:fine-info:address')}</label>
          <p>{foulData.address}</p>
        </div>
        <div className="info-field">
          <label>{t('common:fine-info:additional-details')}</label>
          <p>{foulData.addressAdditionalInfo}</p>
        </div>

        <hr />

        {foulData.fouls.map((foul: Foul, index: number) => (
          <FoulRow key={index} foul={foul} />
        ))}

        <hr />

        <div className="wide-field">
          <label>{t('common:fine-info:fine-additional-details')}</label>
          <p>{foulData.description}</p>
        </div>

        <hr />

        <div className="info-field sum-field">
          <label>{t('common:fine-info:sum')}</label>
          <p>{foulData.invoiceSumText}</p>
        </div>

        <div className="info-field">
          <label>{t('common:fine-info:due-date')}</label>
          <p>{formatDate(foulData.dueDate)}</p>
        </div>

        <hr />
        <Barcode barcode={foulData.barCode} className="wide-field" />
      </div>
    </div>
  );
};

export default ParkingFineSummary;
