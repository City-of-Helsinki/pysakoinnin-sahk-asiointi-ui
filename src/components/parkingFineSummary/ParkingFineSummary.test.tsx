import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import ParkingFineSummary from './ParkingFineSummary';
import mockFoulData from '../../mocks/mockFoulData';
import { Provider } from 'react-redux';
import store from '../../store';

describe('Component', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <ParkingFineSummary foulData={mockFoulData} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('passes A11y checks', async () => {
    const { container } = render(
      <Provider store={store}>
        <ParkingFineSummary foulData={mockFoulData} />
      </Provider>
    );

    expect(await axe(container)).toHaveNoViolations();
  });
});
