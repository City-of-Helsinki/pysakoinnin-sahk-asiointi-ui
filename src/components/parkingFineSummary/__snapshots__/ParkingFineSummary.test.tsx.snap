// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<div>
  <div
    class="summary-container"
  >
    <div
      class="Card-module_card__1WbKx card_hds-card--border__c2dBc vehicle-details-container"
      role="region"
    >
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="regNumber"
        >
          common:fine-info:reg-number:label
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="regNumber"
            readonly=""
            type="text"
            value="common:fine-info:reg-number:placeholder"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="vehicleType"
        >
          parking-fine:vehicle-info:type:label
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="vehicleType"
            readonly=""
            type="text"
            value="parking-fine:vehicle-info:type:placeholder"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="vehicleBrand"
        >
          parking-fine:vehicle-info:brand:label
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="vehicleBrand"
            readonly=""
            type="text"
            value="parking-fine:vehicle-info:brand:placeholder"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="vehicleModel"
        >
          parking-fine:vehicle-info:model:label
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="vehicleModel"
            readonly=""
            type="text"
            value="parking-fine:vehicle-info:model:placeholder"
          />
        </div>
      </div>
      <div
        class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
      >
        <label
          class="FieldLabel-module_label__1zrXK "
          for="vehicleColor"
        >
          parking-fine:vehicle-info:color:label
        </label>
        <div
          class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
        >
          <input
            class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
            id="vehicleColor"
            readonly=""
            type="text"
            value="parking-fine:vehicle-info:color:placeholder"
          />
        </div>
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="fineTimeStamp"
      >
        common:fine-info:fine-timestamp:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="fineTimeStamp"
          readonly=""
          type="text"
          value="common:fine-info:fine-timestamp:placeholder"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="refNumber"
      >
        common:fine-info:ref-number:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="refNumber"
          readonly=""
          type="text"
          value="common:fine-info:ref-number:placeholder"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="address"
      >
        common:fine-info:address:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="address"
          readonly=""
          type="text"
          value="common:fine-info:address:placeholder"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="additionalDetails"
      >
        common:fine-info:additional-details:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="additionalDetails"
          readonly=""
          type="text"
          value="common:fine-info:additional-details:placeholder"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="fineTitle-1"
      >
        common:fine-info:fine-title:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <textarea
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="fineTitle-1"
          readonly=""
        >
          common:fine-info:fine-title:placeholder
        </textarea>
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="fineDetails-1"
      >
        common:fine-info:fine-details:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="fineDetails-1"
          readonly=""
          type="text"
          value="common:fine-info:fine-details:placeholder"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="fineTitle-2"
      >
        common:fine-info:fine-title:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <textarea
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="fineTitle-2"
          readonly=""
        >
          common:fine-info:fine-title:placeholder
        </textarea>
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="fineDetails-2"
      >
        common:fine-info:fine-details:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="fineDetails-2"
          readonly=""
          type="text"
          value="common:fine-info:fine-details:placeholder"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq wide-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="fineAdditionalDetails"
      >
        common:fine-info:fine-additional-details:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <textarea
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="fineAdditionalDetails"
          readonly=""
        >
          common:fine-info:fine-additional-details:placeholder
        </textarea>
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="sum"
      >
        common:fine-info:sum:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="sum"
          readonly=""
          type="text"
          value="common:fine-info:sum:placeholder"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq info-field"
    >
      <label
        class="FieldLabel-module_label__1zrXK "
        for="dueDate"
      >
        common:fine-info:due-date:label
      </label>
      <div
        class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
      >
        <input
          class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
          id="dueDate"
          readonly=""
          type="text"
          value="common:fine-info:due-date:placeholder"
        />
      </div>
    </div>
  </div>
  <div
    class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
  >
    <label
      class="FieldLabel-module_label__1zrXK "
      for="fineBarcode"
    >
      common:fine-info:barcode:label
    </label>
    <div
      class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
    >
      <input
        class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
        id="fineBarcode"
        readonly=""
        type="text"
        value="common:fine-info:barcode:placeholder"
      />
    </div>
  </div>
  <button
    class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS"
    type="button"
  >
    <div
      aria-hidden="true"
      class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
    >
      <svg
        aria-hidden="true"
        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <rect
            height="24"
            width="24"
          />
          <path
            d="M6,10 L6,12 L5,12 L5,18 L12,18 L12,17 L14,17 L14,19 C14,19.5522847 13.5522847,20 13,20 L4,20 C3.44771525,20 3,19.5522847 3,19 L3,11 C3,10.4477153 3.44771525,10 4,10 L6,10 Z M20,4 C20.5522847,4 21,4.44771525 21,5 L21,15 C21,15.5522847 20.5522847,16 20,16 L8,16 C7.44771525,16 7,15.5522847 7,15 L7,5 C7,4.44771525 7.44771525,4 8,4 L20,4 Z M19,6 L9,6 L9,14 L19,14 L19,6 Z"
            fill="currentColor"
          />
        </g>
      </svg>
    </div>
    <span
      class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
    >
      common:copy-barcode
    </span>
  </button>
</div>
`;
